default/python/AI/FreeOrionAI.py

```diff
@@ -138,9 +138,12 @@ def prepareForSave():  # pylint: disable=invalid-name
     print "Preparing for game save by serializing state"
 
     # serialize (convert to string) global state dictionary and send to AI client to be stored in save file
-    dump_string = pickle.dumps(foAIstate)
-    print "foAIstate pickled to string, about to send to server"
-    fo.setSaveStateString(dump_string)
+    try:
+        dump_string = pickle.dumps(foAIstate)
+        print "foAIstate pickled to string, about to send to server"
+        fo.setSaveStateString(dump_string)
+    except:
+        print "foAIstate unable to picke string"
```

Comment:
Hmm, thinking this through a bit more and perusing code-- this does look like it should stop the referenced crash-on-save problem, and the resulting save-file should be playable, but when loading from that save-file the affected AI will have lost its original aggression level and would then be playing as "Aggressive".  

Seems reasonable at least as a stopgap, but still enough of an issue that I am not sure I would want to make it as nearly invisible as this change would make it.  If instead of a simple print statement we use our print error function, then it will also send the warning message in red letters to the host.  In this case, I think the host would be getting this message every turn so we should probably also make it clear that the resulting save file should still be playable, or maybe even note the aggression issue (or point them to the Github issue?).
```        print_error("foAIstate unable to pickle save-state string; the save file should be playable but the AI may have a different aggression."

Looking and thinking a bit more, I may have a fix for the underlying problem in the referenced issue, which I will try out.  But the changes discussed here could still be helpful improvements even if the immediate need went away from my hopeful fix.